samples/infix/reader.cc

```diff
@@ -216,20 +561,33 @@ namespace
               return err(id, "undefined");
             }
 
-            return Ref << id;
+            // we're just checking refs; we only make a change if we find an
+            // error
+            return NoChange ^ "";
           },
       }};
   }
 }
 
 namespace infix
 {
-  Reader reader()
+  Reader reader(const Config& config)
   {
     return {
```

Comment:
See other similar comment. I am convinced and will look into it.

Reply:
It may not be quite what you imagined, but I've pushed something that might look a bit better.

There's still some amount of configuration passing-around, but it's all done with if statements which makes things a little easier to read. Also, I disable passes when possible, though again perhaps not as many as you might have imagined, I don't know.

I could go and duplicate the code if you insist, but I like how this still have basically everything defined in one place, and you can kinda follow along with which parts are enabled / disabled when looking at the enclosing if statements.

Reply:
Well, me and my big mouth. CI showed why disabling passes will not work well in current-day Trieste: it mis-aligns the WF definitions. Even if technically the pass logic works, the fuzzer will explore a scenario like `<allows X> (conditionally omitted <eliminates X if present>) <does not allow X>`. If the X-eliminator is not needed in a given configuration, the fuzzer doesn't know that and generates X from where it's allowed, it is not eliminated because that pass is gone, and it blows up `<does not allow X>`.

To avoid this, you'd have to programmatically change all the WF conditions to guide the fuzzer properly, which, while technically possible, will make things much messier and is not my first choice. Unless you have something to add, my thought is that disabling passes that eliminate parts of the WF (aka. all the reader passes here) is a something I would rather avoid.

I'm reverting the top-level conditionals in `reader()` but keeping the other ones, because they do not cause that problem.

Reply:
After video call, the state I ended up in was agreed to be fine (key issue was usage of action functions for toggling, it turns out)